src/components/Modals/EditExpertProfileModal.tsx

```diff
@@ -0,0 +1,204 @@
+import React, { useContext, useEffect, useState } from 'react';
+import DialogModalBase from './DialogModalBase';
+import Context from '../../context';
+import { ModalContext } from '../../context/ModalContext';
+import TagInput from '../forms/TagInput';
+
+const accentColor = '#4E6AE6';
+const emailValidationRegex = /^(([^<>()[\]\\.,;:\s@"]+(\.[^<>()[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/;
+
+const EditExpertProfileModal = () => {
+  const userContext = useContext(Context.User);
+  const { user } = userContext;
+  const apiContext = useContext(Context.Api);
+  const modalContext = useContext(ModalContext);
+
+  const [email, setEmail] = useState(
+    user.expertData != null ? user.expertData.contactEmail : ''
+  );
+  const [description, setDescription] = useState(
+    user.expertData != null ? user.expertData.description : ''
+  );
+  const [expertiseTags, setExpertiseTags] = useState(
+    user.expertData != null ? user.expertData.expertiseTags : []
+  );
+  const [visibility, setVisibility] = useState(
+    user.expertData != null ? user.expertData.active : null
+  );
+
+  const [fieldsMissing, setFieldsMissing] = useState(false);
+  const [emailInvalid, setEmailInvalid] = useState(false);
+
+  const [availableTags, setAvailableTags] = useState([]);
+
+  useEffect(() => {
+    apiContext.getUsedExpertTags().then((r) => {
+      setAvailableTags(r.map((expertTag) => expertTag.name));
+    });
+  }, []);
+
+  // const [Avatar, setAvatar] = useState(props.avatar);
+  //
+  // const avatarUploader = useRef(null);
+
+  const resetFields = () => {
+    setEmail(user.expertData != null ? user.expertData.contactEmail : '');
+    setDescription(user.expertData != null ? user.expertData.description : '');
+    setExpertiseTags(
+      user.expertData != null ? user.expertData.expertiseTags : ['']
```

Comment:
@realmayus The empty tag still appears for me. Probably that line need to be changed as well?